modules/core/src/smithy4s/schema/FieldFilter.scala

```diff
@@ -158,10 +163,45 @@ object FieldFilter {
 
   val SkipNonRequiredDefaultValues: FieldFilter = skipNonRequiredDefaultValues
 
-  private case object skipUnsetOptions extends FieldFilter.SkipNonRequired {
-    def compileNonRequired[S, A](field: Field[S, A]): Predicate[A] = { a =>
-      a != None
+  private object IsNoneVisitor extends SchemaVisitor.Default[Predicate] {
+    def default[A]: Predicate[A] = Function.const(false)
+
+    override def biject[A, B](
+        schema: Schema[A],
+        bijection: Bijection[A, B]
+    ): Predicate[B] = {
+      if (schema.hints.has(Nullable)) {
+        // nullables are technically never None
+        Function.const(false)
+      } else {
+        val inner = this(schema)
+        a => inner(bijection.from(a))
+      }
+    }
+
+    override def lazily[A](suspend: Lazy[Schema[A]]): Predicate[A] = {
+      val u = suspend.map(this(_))
```

Comment:
this needs to be a lazy val

Reply:
You sure? It's not eagerly evaluating the lazy (hence the map)

Reply:
add a test to convince me :)

Reply:
should be covered by the other comment, yeah